src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsController.java

```diff
@@ -0,0 +1,129 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.List;
+
+import static ch.wisv.events.utils.ResponseEntityBuilder.createResponseEntity;
+
+/**
+ * SalesScanEventController.
+ */
+@Controller
+@RequestMapping(value = "/sales/stats")
+@PreAuthorize("hasRole('USER')")
+public class SalesStatsController {
+    /** Default return redirect on error. */
+    private static final String ERROR_REDIRECT = "redirect:/sales/stats/";
+
+    /**
+     * EventService.
+     */
+    private final EventService eventService;
+
+    /**
+     * AuthenticationService.
+     */
+    private final AuthenticationService authenticationService;
+
+    /**
+     * SalesService.
+     */
+    private final SalesService salesService;
+
+    /**
+     * SalesScanMainController.
+     *
+     * @param eventService of type EventService.
+     * @param authenticationService of type AuthenticationService.
+     * @param salesService of type SalesService.
+     */
+    @Autowired
+    public SalesStatsController(EventService eventService, AuthenticationService authenticationService, SalesService salesService) {
+        this.eventService = eventService;
+        this.authenticationService = authenticationService;
+        this.salesService = salesService;
+    }
+
+    /**
+     * Index view of the stats page.
+     *
+     * @param model of type Model.
+     * @return String
+     */
+    @GetMapping
+    public String IndexView(Model model) {
+        Customer currentUser = authenticationService.getCurrentCustomer();
+        model.addAttribute("events", salesService.getAllGrantedEventByCustomer(currentUser));
+
+        return "sales/stats/index";
+    }
+
+    /**
+     * Ticket sales view.
+     *
+     * @param model of type Model
+     * @param key    of type String
+     * @return String
+     */
+    @GetMapping("/products/{key}")
+    public String ticketSalesindex(Model model, @PathVariable String key) throws EventNotFoundException {
+        Event event = eventService.getByKey(key);
+
+        model.addAttribute("products", event.getProducts());
+        model.addAttribute("key", key);
+
+        return "sales/stats/products/index";
+    }
+
+    /**
+     * Event sales view.
+     *
+     * @param model of type Model
+     * @param key    of type String
+     * @return String
+     */
+    @GetMapping("/event/{key}")
+    public String eventSalesView(Model model, @PathVariable String key) throws EventNotFoundException{
+        Event event = eventService.getByKey(key);
+        List<Order> orders = salesService.getAllOrdersByEvent(event);
+        double moneyEarned = 0.0;
+        int ticketsSold = 0;
+
+        for(Product product: event.getProducts()){
+            moneyEarned += product.getCost() * product.getSold();
+            ticketsSold += product.getSold();
+        }
+
+        model.addAttribute("orders", orders);
```

Comment:
The orders should not be added to the frontend as now it leaking additional information about the orders. For example, the information about other tickets the customer bought.
The order information is not relevant for the committee members, the names of the participants might be useful.

Reply:
The order is currently used for a few parameters:
```
publicReference
createdAtDate
orderProducts
getAmount
getStatus
```
I will set all data that should not be available to the front-end to null to allow the data to be anonymously used

Reply:
Is this sufficient?